clifford/_layout.py

```diff
@@ -572,12 +572,37 @@ def comp_func(Xval):
             return Yval
         return comp_func
 
+
+    @_cached_property
+    def _shirokov_inverse(self):
+        """
+        Performs the inversion operation as described in Theorem 4, page 16 of the paper :cite:`shirokov2020inverse`
+        """
+        n = len(self.sig)
+        exponent = (n + 1) // 2
+        N = 2 ** exponent
+        @_numba_utils.njit
+        def shirokov_inverse(U):
+            Uk1 = U
+            Uk = Uk1
+            Ck = 0.0
+            for k in range(1, N):
+                Uk = Uk1
+                Ck = (N / k) * Uk.value[0]
+                Uk1 = U * (Uk - Ck)
+            adjU = Ck - Uk
+            detU = -Uk1.value[0]
+            if detU == 0:
+                raise ValueError('Multivector has no inverse')
+            return adjU / detU
```

Comment:
I think this is equivalent:
```suggestion
        def shirokov_inverse(U):
            Uk = U * 1.0  # cast to float
            for k in range(1, N):
                Ck = (N / k) * Uk.value[0]
                adjU = (Uk - Ck)
                Uk = U * adjU
            if Uk.value[0] == 0:
                raise ValueError('Multivector has no inverse')
            return adjU / Uk.value[0]
```
Note  you need the `* 1.0` to prevent numba crashing on integer multivectors.